src/python/utils/diff_statistics.py

```diff
@@ -0,0 +1,167 @@
+from pathlib import Path
+from typing import Union, Set
+
+from loguru import logger
+
+from src.python.repo import Repository
+
+
+def get_diff(
+    repo: Repository, left_sha: str, right_sha: str, diff_log_file: Union[None, Path]
+) -> str:
+    """
+    Computes the diff between two branches using git diff.
+    Args:
+        repo (Repository): The repository object.
+        left_sha (str): The left sha.
+        right_sha (str): The right sha.
+    Returns:
+        str: A string containing the diff result.
+    """
+    command = f"git diff {left_sha} {right_sha}"
+    stdout, _ = repo.run_command(command)
+    return stdout
+
+
+def get_diff_files_branches(repo: Repository, left_sha: str, right_sha: str) -> set:
+    """
+    Computes the diff between two branches using git diff.
+    Args:
+        repo (Repository): The repository object.
+        left_sha (str): The left sha.
+        right_sha (str): The right sha.
+    Returns:
+        set: A set containing the diff result.
+    """
+    # Using git diff to compare the two SHAs
+    command = f"git diff --name-only {left_sha} {right_sha}"
+    stdout, _ = repo.run_command(command)
+    return set(stdout.split("\n")) if stdout else set()
+
+
+def get_diff_hunks(repo: Repository, left_sha: str, right_sha: str) -> int:
+    """
+    Compute the number of hunks that are different between two branches using git diff.
+    Args:
+        repo (Repository): The repository object.
+        left_sha (str): The left sha.
+        right_sha (str): The right sha.
+    Returns:
+        int: The number of hunks that are different between the two branches.
+    """
+    diff, _ = repo.run_command(
+        f"git diff --unified=0 {left_sha} {right_sha} | grep -c '^@@'"
+    )
+    return int(diff)
+
+
+def get_diff_files_merge(
+    repo: Repository,
+    left_sha: str,
+    right_sha: str,
+) -> Set[str]:
+    """
+    Computes the diff between two branches using git diff.
+    Args:
+        repo (Repository): The repository object.
+        merge_idx (str): The merge index, such as 42-123.
+        repo_slug (str): The repository slug.
+        left_sha (str): The left sha.
+        right_sha (str): The right sha.
+        cache_dir (Path): The path to the cache directory.
+    Returns:
+        Set[str]: A set containing the diff result.
+    """
+    command = f"git merge-base {left_sha} {right_sha}"
```

Comment:
The merge base was already computed in the caller, so you could pass it in if you like.  It's not essential, though.